### DISCUSSION TITLE
Git helpers

### DISCUSSION THREAD

**Discussion by dberenbaum** [treeverse/dvc#6929] [General]
Related to https://github.com/iterative/dvc.org/issues/2969.

DVC has two features that reduce a lot of boilerplate/duplicate Git commands:

* `core.autostage`: staging changes to DVC files.
* `install`: install hooks for checkout, commit, and push.

It would be great to make these more obvious and useful to people. There are a few options:

1. Document and promote them more (related to the linked issue above).
2. Make them the default settings. For `core.autostage`, I think it could make sense. For `install`, I think there are some reasons not to do this:
  a. It seems aggressive to install Git hooks when initializing the repo.
  b. There are more use cases where people don't want to couple the DVC and Git actions (for example, it's expensive to push to my remote and I want to wait until the end of the day to do it).
  c. It hides more from the user. Understanding `dvc checkout` and other commands are helpful for beginners.
3. Prompt when doing `dvc init` whether to turn them on.

**Comment by daavoo** [comment]
We are currently prompting in `dvc exp init -i`. 

If we include dvc initialization as part of that command, we could prompt for the `autostage` and `install`?

Additionally (this goes back to old "prompt / not prompt" discussion) we could introduce a -i flag for `dvc init`.

**Reply by dberenbaum** [reply]
Interesting idea. Right now we are staging everything from `dvc exp init` but not setting `core.autostage`. We could set that, but I'm not sure it makes sense as part of that command, and it might become too noisy. What do you think, @skshetry?
